app/controllers/charges_controller.rb

```diff
@@ -21,39 +20,40 @@ def create
     amount = line_items.inject(0) { |sum, item| sum + item['amount'] * item['quantity'] }
 
     description = generate_description(
-      seller_names: seller_names.to_a,
-      item_types: item_types
-    )
-    intent = Stripe::PaymentIntent.create(
-      amount: amount,
-      currency: 'usd',
-      receipt_email: charge_params[:email],
-      payment_method_types: ['card'],
-      description: description
+        seller_names: seller_names.to_a,
+        item_types: item_types
     )
 
-    # Creates a pending PaymentIntent. See webhooks_controller to see what happens
-    # when the PaymentIntent is successful.
-    PaymentIntent.create!(
-      stripe_id: intent['id'],
-      email: charge_params[:email],
-      line_items: line_items.to_json
-    )
+    email = charge_params[:email]
+    if ENV['USE_SQUARE'] == "true"
+      payment = create_square_payment_request(
+          source_id: charge_params[:nonce],
+          amount: amount,
+          note: description,
+          email: email,
+          line_items: line_items)
+    else
+      payment = create_stripe_payment_request(
+          amount: amount,
+          email: email,
+          description: description,
+          line_items: line_items)
+    end
 
-    json_response(intent)
+    json_response(payment)
   end
 
   private
 
   def charge_params
     params.require(:line_items)
     params.require(:email)
-    params.permit(:email, line_items: [[:amount, :currency, :item_type, :quantity, :seller_id]])
+    params.permit(:email, :nonce, line_items: [[:amount, :currency, :item_type, :quantity, :seller_id]])
   end
 
   def validate(line_item:)
     [:amount, :currency, :item_type, :quantity, :seller_id].each do |attribute|
-    unless line_item.key?(attribute)
+      unless line_item.key?(attribute)
```

Comment:
I think this tab is supposed to be here

Reply:
Sorry did you mean you don't think the tab is supposed to be here? I think my auto-formatter did this